ls1intum/Artemis
# `Development`: Refactor DataExportCreationService and add Art. 15 GDPR README

### Checklist
#### General


#### Server

### Motivation and Context


The DataExportCreationService has too many dependencies and is overly complex with > 850 LOC
To improve code quality, this service should be refactored.
Additionally, the data export requires a README, see #6851 

### Description

Split this service up into 5  and moved all related service into their own `dataexport` package.

- `DataExportCreationService`: handles the core logic, relies on the other ones
- `DataExportExerciseCreationService`: creates the data for exercise participations
- `DataExportExamCreationService`creates the data for exam participations
- `DataExportQuizExerciseCreationService`: creates the data for quiz exercises (since the quiz exercise export is rather complex compared to the other exercise types)
-  ` DataExportCommunicationDataService` extracts all the data about posts, messages, reactions a user has made.
Additionally, add a readme file, see #6851 


### Steps for Testing
Code Review for the refactoring:
For the readme:
Prerequisites:

1 User with an email address you have access to
Log in to Artemis
Navigate to Privacy
Navigate to Data Export
Request a data export
Wait until you receive your email
Download the data export
Make sure the zip contains the README.md file

#### Performance Review
#### Code Review

### Manual tests
- [x] Manuel Test 1
- [x] Manuel Test 2

### Test Coverage






#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [DataExportCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportCreationService.html) | 97% | ✅ |
| [DataExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportService.html) | 99% | ✅|
| [DataExportCommunicationDataService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportCommunicationDataService.html) | 99% | ✅ |
| [DataExportCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportCreationService.html) | 97% | ✅ |
| [DataExportExamCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportExamCreationService.html) | 97% | ✅|
| [DataExportExerciseCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportExerciseCreationService.html) | 90% | ✅ |
| [DataExportQuizExerciseCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportQuizExerciseCreationService.html) | 100% | ✅ |
| [DataExportUtil.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportUtil.html) | 100% | ✅ |
| [DataExportScheduleService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.scheduled/DataExportScheduleService.html) | 95% | ✅ |
| [DataExportResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/DataExportResource.html) | 98% | ✅ |

### Screenshots
